### PR TITLE
Update Controls.cs

### PR DESCRIPTION
**Pull request by reggaeshark24**
changed how the character gets velocity; added a isgrounded function (needs new boxcollider2d as shown in picture)
boxcollider2d settings

### CODE REVIEW

File: Take 2.1/Assets/Scripts/Controls.cs [hunk 1]
```diff
@@ -27,8 +30,11 @@ public class Controls : MonoBehaviour
     void Start()
     {
         //grabbing components
+        moveSpeed = 0.3f;
+        jumpForce = 8f;
+        isJumping = false;
         animator = GetComponent<Animator>();
-        rb = GetComponent<Rigidbody2D>();
+        rb = gameObject.GetComponent<Rigidbody2D>();
```
Comment (Lunnight) [inline Take 2.1/Assets/Scripts/Controls.cs:37]: Is the "gameObject" bit necessary if it's just referring to the player character as this script is attached to it

File: Take 2.1/Assets/Scripts/Controls.cs [hunk 2]
```diff
@@ -37,13 +43,14 @@ void Start()
     void Update()
     {
         //grabs horizontal movement input
-        inputX = Input.GetAxis("Horizontal");
+        inputX = Input.GetAxisRaw("Horizontal");
+        inputY = Input.GetAxisRaw("Vertical");
 
-        if (Input.GetKeyDown("space"))
+        if (Input.GetKeyDown("f"))
```
Comment (Lunnight) [inline Take 2.1/Assets/Scripts/Controls.cs:49]: I thought we decided on changing the attack key to enter?

File: Take 2.1/Assets/Scripts/Controls.cs [hunk 3]
```diff
@@ -57,13 +64,21 @@ void Update()
     private void FixedUpdate()
     {
         //flips sprite on x-axis when changing direction
-        if (inputX > 0)
-            transform.localScale = new Vector3(-1.0f, 1.0f, 1.0f);
-        if (inputX < 0)
+        if (inputX > 0.1f)
             transform.localScale = new Vector3(1.0f, 1.0f, 1.0f);
+        if (inputX < 0.1f)
```
Comment (Lunnight) [inline Take 2.1/Assets/Scripts/Controls.cs:69]: doesn't this mean that the player will move left if nothing is being input as doesn't 'getAxisRaw' produce either: = -1, 0 or 1?

File: Take 2.1/Assets/Scripts/Controls.cs [hunk 4]
```diff
@@ -57,13 +64,21 @@ void Update()
...
+            transform.localScale = new Vector3(-1.0f, 1.0f, 1.0f);
 
         //give character velocity to move
-        rb.velocity = new Vector2(inputX * speed, rb.velocity.y);
+        if(inputX > 0.1f || inputX < -0.1f)
```
Comment (Lunnight) [inline Take 2.1/Assets/Scripts/Controls.cs:73]: what do the two || mean?

File: Take 2.1/Assets/Scripts/Controls.cs (hunk 1 above)
Comment (reggaeshark24) [reply Take 2.1/Assets/Scripts/Controls.cs:37]: It's not, that's just how I re-wrote it when moved down a few lines

File: Take 2.1/Assets/Scripts/Controls.cs [hunk 5]
```diff
@@ -57,13 +64,21 @@ void Update()
...
+        {
+          rb.AddForce(new Vector2(inputX * moveSpeed, 0f), ForceMode2D.Impulse);
+        }
+        if(!isJumping && inputY > 0.1f)
```
Comment (Lunnight) [inline Take 2.1/Assets/Scripts/Controls.cs:77]: need to then change isJumping to true. then change it back when it hits the ground again

**Review by Lunnight — COMMENTED** [review]
Just need these couple of things to be checked then will test it

File: Take 2.1/Assets/Scripts/Controls.cs (hunk 4 above)
Comment (reggaeshark24) [reply Take 2.1/Assets/Scripts/Controls.cs:73]: || means OR

File: Take 2.1/Assets/Scripts/Controls.cs (hunk 1 above)
Comment (Lunnight) [reply Take 2.1/Assets/Scripts/Controls.cs:37]: aah kk

File: Take 2.1/Assets/Scripts/Controls.cs (hunk 4 above)
Comment (Lunnight) [reply Take 2.1/Assets/Scripts/Controls.cs:73]: oh. lol I dumb

File: Take 2.1/Assets/Scripts/Controls.cs (hunk 2 above)
Comment (reggaeshark24) [reply Take 2.1/Assets/Scripts/Controls.cs:49]: I forgot about this ^

Comment (Lunnight) [reply Take 2.1/Assets/Scripts/Controls.cs:49]: ok now worries, I can change that

**Review by Lunnight — APPROVED** [review]
Ok will change the key for attack and test, and add health

**Pull request merged**